### ISSUE TITLE
Direct3D applications crashing when installing the D3D package

### ISSUE THREAD

**Issue by JW26T-Prj** [Skulltrail192/One-Core-API-Binaries#173]
When I install the D3D runtime package, any application that uses Direct3D will crash. I tried changing the video driver and no success at all.

One-Core-API Release 3.0.0
CPU: Intel(R) Core(TM)2 Quad  CPU   Q9550  @ 2.83GHz
Motherboard: Gigabyte GA-EQ45M-S2 (rev 2.0)
RAM: 8GB DDR2 800MHz
HDD: 320GB SATA2
Graphics: Integrated Intel(R) Q45/Q43 Express Chipset
OS: Windows XP Professional SP3 32-bit pt-BR (because other languages doesn't boot on my machine)

**Comment by jonm58** [comment]
What is your order of installation?
I use VLC and Direct3D works fine.

**Comment by JW26T-Prj** [comment]
I used the Base first: Base-> API-SET -> Branding -> D3d -> Modern Setup -> Driver Update -> App Compat-> Additional Dlls;

**Comment by jonm58** [comment]
> I used the Base first: Base-> API-SET -> Branding -> D3d -> Modern Setup -> Driver Update -> App Compat-> Additional Dlls;

I used the Base first,No Installation Modern Setup
Base-> API-SET -> Branding -> D3d -> Driver Update -> App Compat-> Additional Dlls;

**Comment by JW26T-Prj** [comment]
I will try soon

**Comment by JW26T-Prj** [comment]
It doesn't work, still getting errors with Direct3D applications

**Comment by jonm58** [comment]
> It doesn't work, still getting errors with Direct3D applications

so,this  applications is?

**Comment by JW26T-Prj** [comment]
basically any application that use Direct3D, i tried with Media Player Classic and some games

**Comment by jonm58** [comment]
![Windows XP Professional-2023-08-13-09-36-02](https://github.com/Skulltrail192/One-Core-API-Binaries/assets/73036321/edb8015c-cb49-4ba7-b4d5-d537811b85f5)
maybe you need this:
Windows XP Professional SP3 x86 - Integral Edition[https://zone94.com/downloads/software/operating-systems/123-windows-xp-professional-sp3-x86-integral-edition](url)

**Comment by Skulltrail192** [comment]
Version 2.10 these applications works?

**Comment by JW26T-Prj** [comment]
I didn't tried with 2.10.0, but will check soon

**Comment by JW26T-Prj** [comment]
> ![Windows XP Professional-2023-08-13-09-36-02](https://github.com/Skulltrail192/One-Core-API-Binaries/assets/73036321/edb8015c-cb49-4ba7-b4d5-d537811b85f5)
> maybe you need this:
> Windows XP Professional SP3 x86 - Integral Edition[https://zone94.com/downloads/software/operating-systems/123-windows-xp-professional-sp3-x86-integral-edition](url)
> 
> edit:Maybe you need to update the POS2009 patch
> 

I tried installing One-Core 3.0.0 with this XP Iso and still getting the same issues with Direct3D

Same issues with version 2.10.0

**Comment by Stepman123** [comment]
> When I install the D3D runtime package, any application that uses Direct3D will crash. I tried changing the video driver and no success at all.

When you set "One Core API", your application goes into DirectX 10/11 mode, but D3d's "One Core API" supports **not any** applications.
 If the application supports DirectX 9, you can try to force this mode.  For example, in the game S.T.A.L.K.E.R, for this you need to remove two .dll.

**Comment by JW26T-Prj** [comment]
I understand...

There is a way to force DirectX 9 for all applications?

**Comment by Skulltrail192** [comment]
Well, for now, One-Core-API use WineD3D to give DirectX 10/11 support. However, WineD3d not run fine on InteL Graphics and maybe run wrong on AMD Radeon. So, maybe this is the issue. Directx 9 is not affected, because run native.

**Comment by JW26T-Prj** [comment]
Yes, i'm using a Intel GMA 4500 integrated graphics (that have DirectX 10 support).

**Comment by jonm58** [comment]
> Yes, i'm using a Intel GMA 4500 integrated graphics (that have DirectX 10 support). Maybe a variant of D3D runtime can be released without WineD3D?
uh,you can try
Base-> API-SET -> Branding -> Driver Update -> App Compat-> Additional Dlls;

**Comment by Skulltrail192** [comment]
Unfortunatly, i doesn't have other directx option. If vulkan was released for XP, we would directx 10,11 and 12 now. I have plans to getting WDDM, however, is very early and not work and make a custom directx, but, need time.

**Comment by Stepman123** [comment]
> Unfortunatly, i doesn't have other directx option. If vulkan was released for XP, we would directx 10,11 and 12 now. I have plans to getting WDDM, however, is very early and not work and make a custom directx, but, need time.

The Mesa3D project may be useful. https://github.com/pal1000/mesa-dist-win/releases/tag/17.1.8.401-1
I used DLL from the newest version for launching "GPU Caps Viewer" https://www.geeks3d.com/dl/show/585
As a result, I have supported GL 3-4 partially, albeit without hardware acceleration. But for those who use virtual machines on modern equipment, hardware acceleration will work, as they say in the Issues of the Mesa3D project.
[](http://ipic.su/img/img7/fs/GL4.1692151180.jpg)

**Comment by jonm58** [comment]
![image](https://github.com/Skulltrail192/One-Core-API-Binaries/assets/73036321/6ab7663d-28c5-403d-9adf-d960ff0d378a)

**Comment by Skulltrail192** [comment]
> ![image](https://user-images.githubusercontent.com/73036321/263578861-6ab7663d-28c5-403d-9adf-d960ff0d378a.png)

I see, it is implemented by _Dark_Fire, a reactos's developer and youtuber.
